// @routerprotocol/router-crosstalk/typechain/factories/IERCHandler__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type { IERCHandler, IERCHandlerInterface } from "../IERCHandler";

const _abi = [
  {
    inputs: [],
    name: "_reserve",
    outputs: [
      {
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "uint8",
        name: "destinationChainID",
        type: "uint8",
      },
      {
        internalType: "address",
        name: "feeTokenAddress",
        type: "address",
      },
    ],
    name: "getBridgeFee",
    outputs: [
      {
        internalType: "uint256",
        name: "",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "",
        type: "uint256",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
];

export class IERCHandler__factory {
  static readonly abi = _abi;
  static createInterface(): IERCHandlerInterface {
    return new utils.Interface(_abi) as IERCHandlerInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): IERCHandler {
    return new Contract(address, _abi, signerOrProvider) as IERCHandler;
  }
}
