// @routerprotocol/router-crosstalk/typechain/factories/ISequencerHandler__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  ISequencerHandler,
  ISequencerHandlerInterface,
} from "../ISequencerHandler";

const _abi = [
  {
    inputs: [
      {
        components: [
          {
            internalType: "address",
            name: "_rSyncContract",
            type: "address",
          },
          {
            internalType: "uint8",
            name: "_chainID",
            type: "uint8",
          },
          {
            internalType: "address",
            name: "_linkedContract",
            type: "address",
          },
        ],
        internalType: "struct iSequencerHandler.RouterLinker",
        name: "linker",
        type: "tuple",
      },
    ],
    name: "MapContract",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        components: [
          {
            internalType: "address",
            name: "_rSyncContract",
            type: "address",
          },
          {
            internalType: "uint8",
            name: "_chainID",
            type: "uint8",
          },
          {
            internalType: "address",
            name: "_linkedContract",
            type: "address",
          },
        ],
        internalType: "struct iSequencerHandler.RouterLinker",
        name: "linker",
        type: "tuple",
      },
    ],
    name: "UnMapContract",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "bytes",
        name: "_data",
        type: "bytes",
      },
    ],
    name: "executeProposal",
    outputs: [
      {
        internalType: "bool",
        name: "",
        type: "bool",
      },
    ],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [],
    name: "fetch_chainID",
    outputs: [
      {
        internalType: "uint8",
        name: "",
        type: "uint8",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "uint8",
        name: "_destChainID",
        type: "uint8",
      },
      {
        internalType: "bytes",
        name: "_generic",
        type: "bytes",
      },
      {
        internalType: "uint256",
        name: "_gasLimit",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "_gasPrice",
        type: "uint256",
      },
      {
        internalType: "address",
        name: "_feeToken",
        type: "address",
      },
    ],
    name: "genericDeposit",
    outputs: [
      {
        internalType: "uint64",
        name: "",
        type: "uint64",
      },
    ],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "uint8",
        name: "_destChainID",
        type: "uint8",
      },
      {
        internalType: "bytes",
        name: "_erc20",
        type: "bytes",
      },
      {
        internalType: "bytes",
        name: "_swapData",
        type: "bytes",
      },
      {
        internalType: "bytes",
        name: "_generic",
        type: "bytes",
      },
      {
        internalType: "uint256",
        name: "_gasLimit",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "_gasPrice",
        type: "uint256",
      },
      {
        internalType: "address",
        name: "_feeToken",
        type: "address",
      },
      {
        internalType: "bool",
        name: "_isTransferFirst",
        type: "bool",
      },
    ],
    name: "genericDepositWithERC",
    outputs: [
      {
        internalType: "uint64",
        name: "",
        type: "uint64",
      },
    ],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "uint8",
        name: "_destChainID",
        type: "uint8",
      },
      {
        internalType: "uint64",
        name: "_depositNonce",
        type: "uint64",
      },
      {
        internalType: "uint256",
        name: "_gasLimit",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "_gasPrice",
        type: "uint256",
      },
    ],
    name: "replayDeposit",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];

export class ISequencerHandler__factory {
  static readonly abi = _abi;
  static createInterface(): ISequencerHandlerInterface {
    return new utils.Interface(_abi) as ISequencerHandlerInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): ISequencerHandler {
    return new Contract(address, _abi, signerOrProvider) as ISequencerHandler;
  }
}
